tests/test_transform_dependency.py

```diff
@@ -0,0 +1,629 @@
+        # we need to invoke the transformation on the full source file and activate recursion to contained nodes
+        kernel.apply(transformation, role='kernel')
+        driver['driver'].apply(transformation, role='driver', targets=('kernel', 'some_const'))
+
+    # Check that the global variable declaration remains unchanged
+    assert kernel.modules[0].variables[0].name == 'some_const'
+
+    # Check that calls and matching import have been diverted to the re-generated routine
+    calls = FindNodes(CallStatement).visit(driver['driver'].body)
+    assert len(calls) == 1
+    assert calls[0].name == 'kernel_test'
+    imports = FindNodes(Import).visit(driver['driver'].spec)
+    assert len(imports) == 2
+    assert isinstance(imports[0], Import)
+    assert driver['driver'].spec.body[0].module == 'kernel_test_mod'
+    assert 'kernel_test' in [str(s) for s in driver['driver'].spec.body[0].symbols]
+
+    # Check that global variable import remains unchanged
+    assert isinstance(imports[1], Import)
+    assert driver['driver'].spec.body[1].module == 'kernel_mod'
+    assert 'some_const' in [str(s) for s in driver['driver'].spec.body[1].symbols]
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+@pytest.mark.parametrize('use_scheduler', [False, True])
+def test_dependency_transformation_globalvar_imports_driver_mod(frontend, use_scheduler, tempdir, config):
+    """
+    Test that global variable imports are not renamed as a
+    call statement would be.
+    """
+
+    kernel_fcode = """
+MODULE kernel_mod
+    INTEGER :: some_const
+CONTAINS
+    SUBROUTINE kernel(a, b, c)
+    INTEGER, INTENT(INOUT) :: a, b, c
+
+    a = 1
+    b = 2
+    c = 3
+  END SUBROUTINE kernel
+END MODULE kernel_mod
+    """.strip()
+
+    driver_fcode = """
+MODULE DRIVER_MOD
+    USE kernel_mod, only: kernel
+    USE kernel_mod, only: some_const
+CONTAINS
+SUBROUTINE driver(a, b, c)
+    INTEGER, INTENT(INOUT) :: a, b, c
+
+    CALL kernel(a, b ,c)
+END SUBROUTINE driver
+END MODULE DRIVER_MOD
+    """.strip()
+
+    transformation = DependencyTransformation(suffix='_test', module_suffix='_mod')
+
+    if use_scheduler:
+        (tempdir/'kernel_mod.F90').write_text(kernel_fcode)
+        (tempdir/'driver_mod.F90').write_text(driver_fcode)
+        scheduler = Scheduler(paths=[tempdir], config=SchedulerConfig.from_dict(config), frontend=frontend)
+        scheduler.process(transformation, use_file_graph=True)
+
+        kernel = scheduler['kernel_mod#kernel'].source
+        driver = scheduler['driver_mod#driver'].source
+
+    else:
+        kernel = Sourcefile.from_source(kernel_fcode, frontend=frontend)
+        driver = Sourcefile.from_source(driver_fcode, frontend=frontend)
+
+        # Because the renaming is intended to be applied to the routines as well as the enclosing module,
+        # we need to invoke the transformation on the full source file and activate recursion to contained nodes
```

Comment:
With the removal of the `recurse_to_contained_nodes` option, this comment seems outdated. It's also repeated a few times throughout the test file.

Reply:
Yes! I removed the branching in the test related to this but overlooked the comments. Will fix